drivers/src/idevid_csr.rs

```diff
@@ -0,0 +1,49 @@
+// Licensed under the Apache-2.0 license
+
+use core::mem::size_of;
+use zerocopy::{AsBytes, FromBytes};
+use zeroize::Zeroize;
+
+use crate::memory_layout;
+
+pub const MAX_CSR_SIZE: usize = 512;
+
+#[derive(FromBytes, AsBytes, Zeroize)]
+#[repr(C)]
+pub struct IDevIDCsr {
```

Comment:
nit: Should this struct go in the persistent data structure since it's really only intended to be used there?

Reply:
Done. I'll wait to push these changes until the UART logs PR is merged.